src/test/tests/mavenUtils.test.ts

```diff
@@ -0,0 +1,396 @@
+import { before } from 'mocha';
+import * as path from 'path';
+import * as Collections from 'typescript-collections';
+import * as vscode from 'vscode';
+import { ConnectionManager } from '../../main/connect/connectionManager';
+import { LogManager } from '../../main/log/logManager';
+import { ScanCacheManager } from '../../main/scanCache/scanCacheManager';
+import { DependenciesTreeNode } from '../../main/treeDataProviders/dependenciesTree/dependenciesTreeNode';
+import { TreesManager } from '../../main/treeDataProviders/treesManager';
+import { assert } from 'chai';
+import { ComponentDetails } from 'xray-client-js';
+import { GavGeneralInfo } from '../../main/types/gavGeneralinfo';
+import { GeneralInfo } from '../../main/types/generalInfo';
+import { MavenTreeNode } from '../../main/treeDataProviders/dependenciesTree/mavenTreeNode';
+import { PomTree } from '../../main/utils/pomTree';
+import { MavenUtils } from '../../main/utils/mavenUtils';
+import { ContextUtils } from '../../main/utils/contextUtils';
+
+/**
+ * Test functionality of @class NpmUtils.
+ */
+describe('Maven Utils Tests', () => {
+    let dummyScanCacheManager: ScanCacheManager = new ScanCacheManager().activate({
+        workspaceState: { get(key: string) {} } as vscode.Memento
+    } as vscode.ExtensionContext);
+    let treesManager: TreesManager = new TreesManager(
+        [],
+        new ConnectionManager(),
+        dummyScanCacheManager,
+        new LogManager().activate({} as vscode.ExtensionContext)
+    );
+    let dummyProgress: vscode.Progress<{ message?: string; increment?: number }> = { report: () => {} };
+    let projectDirs: string[] = ['dependency', 'empty', 'multiPomDependency'];
+    let workspaceFolders: vscode.WorkspaceFolder[];
+    let tmpDir: vscode.Uri = vscode.Uri.file(path.join(__dirname, '..', 'resources', 'maven'));
+    let outputPath: vscode.Uri = vscode.Uri.file(path.join(tmpDir.fsPath, 'testdata', '1'));
+    const multiDep: string | undefined = ContextUtils.readFileIfExists(outputPath.fsPath);
```

Comment:
This is not make any sense to create a `file` from path and take only the path.
Please consider this refactor:
```typescript
let tmpDir: vscode.Uri = vscode.Uri.file(path.join(__dirname, '..', 'resources', 'maven'));
let outputPath: string = path.join(tmpDir.fsPath, 'testdata', '1');
const multiDep: string | undefined = ContextUtils.readFileIfExists(outputPath);
outputPath = path.join(tmpDir.fsPath, 'testdata', '2');
const multi1Dep: string | undefined = ContextUtils.readFileIfExists(outputPath);
outputPath = path.join(tmpDir.fsPath, 'testdata', '3');
const multi2Dep: string | undefined = ContextUtils.readFileIfExists(outputPath);
outputPath = path.join(tmpDir.fsPath, 'testdata', '4');
const multi3Dep: string | undefined = ContextUtils.readFileIfExists(outputPath);
```